src/utils/first.ts

```diff
@@ -1,11 +1,11 @@
 // Return the first non-null or -undefined result from an array of
 // maybe-sync, maybe-promise-returning functions
 export default function first<T> (candidates: ((...args: any[]) => Promise<T | void> | T | void)[]): (...args: any[]) => Promise<T | void> {
-	return function (...args: any[]) {
+	return function (this: any, ...args: any[]) {
 		return candidates.reduce((promise, candidate) => {
 			return promise.then(
 				result =>
-					result != null ? result : Promise.resolve(candidate(...args))
+					result != null ? result : Promise.resolve(candidate.apply(this, args))
```

Comment:
Are these changes intended?

Reply:
It comes from #1820 (which should be merged first IMO). It is so plugins that also do realpath (multi-entry, node-resolve) can read the option via `this.preserveSymlinks` and act accordingly, instead of each having their own duplicate options.

Reply:
It's also needed for https://github.com/rollup/rollup/pull/1819/files#diff-9b4fdd2a9bd0304295b89fa6270add84R54 to have `this` context.